### ISSUE TITLE
Unable to build page

### ISSUE THREAD

**Issue by xiangzhu** [workflowr/workflowr#70]
Hi John,

I guess this is more of a GitHub issue than a `workflowr` issue. I wonder whether the only solution would be "Please try again later", when this type of error happens. Is there any alternative solution?

In addition, does "try again" mean that I should `git push` again?

Thanks for your time and help! 

```
The page build failed for the `master` branch with the following error:

unable to build page. Please try again later.

For information on troubleshooting Jekyll see:

  https://help.github.com/articles/troubleshooting-jekyll-builds

If you have any questions you can contact us by replying to this email.
```

**Comment by pcarbo** [comment]
@xiangzhu I've gotten messages like this before. It is frustrating because the messages don't give you very much information. Do you have any Markdown (`.md`) files that Jekyll might be trying to build into webpages?

**Comment by xiangzhu** [comment]
@pcarbo  Now I am more than sure that this may be a pure GitHub issue: I tried to push some commits again, the error disappeared, and the webpage was updated as expected.

This is the first time I saw this type of error. Perhaps it will be helpful to mention this possibility somewhere in the documentation?

**Comment by jdblischak** [comment]
@xiangzhu Thanks for reporting this. I think you are right that this is a GitHub Pages issue. Sometime the service is a bit slow to respond, but for a free service I find it quite reliable overall.

>  I tried to push some commits again, the error disappeared, and the webpage was updated as expected.

Could you please post the workflowr GitHub repo and the specific analysis file you added when you first got this error? I'd like to rule out that there was any issue that could have caused this.

> This is the first time I saw this type of error. Perhaps it will be helpful to mention this possibility somewhere in the documentation?

It's hard for me to anticipate all the things that could go wrong on the GitHub servers. What sort of message in the documentation would have been useful to you? Furthermore, where in the documentation would you have been likely to look? The FAQ?

Currently I have the following [footnote](
https://jdblischak.github.io/workflowr/articles/wflow-01-getting-started.html#fn5) about GitHub Pages in the Getting Started vignette:

> It may take a few minutes for the site to be rendered.

**Comment by pcarbo** [comment]
@jdblischak I would simply refer to the help for GitHub Pages if users encounter any errors. I've experienced issues with the Github Pages server once or twice and after waiting ~30 minutes it started to work again.

**Comment by xiangzhu** [comment]
@jdblischak  Below are my replies to your questions.

> Could you please post the workflowr GitHub repo and the specific analysis file you added when you first got this error? I'd like to rule out that there was any issue that could have caused this.

https://github.com/xiangzhu/stanford-stats110/blob/master/analysis/week1_review.Rmd

> What sort of message in the documentation would have been useful to you? Furthermore, where in the documentation would you have been likely to look? The FAQ?

Perhaps let the users know that this type of error may happen, mainly due to issues of GitHub Pages, rather than `workflowr`. When I saw the error for the first time, my first reaction was to ask myself "did I do anything wrong when using `workflowr`"?

I think FAQ is a good idea. We can place it under a section that lists all possible issues that are not caused by `workflowr` itself.

**Comment by jdblischak** [comment]
> https://github.com/xiangzhu/stanford-stats110/blob/master/analysis/week1_review.Rmd

I can't view that page because the repo is private. I was able to view the website and confirm that you have the `.nojekyll` file, which prevents Jekyll from trying to parse the files.

https://xiangzhu.github.io/stanford-stats110/.nojekyll

> I think FAQ is a good idea. We can place it under a section that lists all possible issues that are not caused by workflowr itself.

I added an FAQ entry in https://github.com/jdblischak/workflowr/commit/cc825033be6016a0c132fd326041e4baf69806c2

**Comment by xiangzhu** [comment]
@jdblischak  Sorry the repo is private because I am also using it to organize sensitive documents such as the final exam draft and homework solutions.

The FAQ entry looks good to me! Do you think it will be helpful to add the following example of email from GitHub for this type of issue?

```
The page build failed for the `master` branch with the following error:

unable to build page. Please try again later.

For information on troubleshooting Jekyll see:

  https://help.github.com/articles/troubleshooting-jekyll-builds

If you have any questions you can contact us by replying to this email.
```

I think I do not have any problem regarding this issue, and I will close the issue. Thanks again!

**Comment by jdblischak** [comment]
> Do you think it will be helpful to add the following example of email from GitHub for this type of issue?

@xiangzhu Good idea. I missed that this was an email. Added in https://github.com/jdblischak/workflowr/commit/c81b8d1138927b9e913925877b0c180adbeb738a

**Comment by floribon** [comment]
Hey in case that help, I just ended up here instead of the FAQ from a web search.

Anyway, it seem that in my case I was responsible for it. I got the error twice while pushing after sending a first commit.

I double checked that commit causing it and a git status would display `added:    "some_path/Icon\r"` while I didn't see this file on my file system.

I have no idea where it comes from, but the fact that it contained a `\r` character on its name was likely to cause issues.

I deleted this file and force pushed and the error was gone.
